src/controller.cpp

```diff
@@ -539,26 +539,25 @@ void controller::mark_all_read(const std::string& feedurl)
 		return;
 	}
 
-	std::lock_guard<std::mutex> feedslock(feeds_mutex);
-	for (const auto& feed : feedcontainer.feeds) {
-		std::lock_guard<std::mutex> lock(feed->item_mutex);
-
-		if (feedurl.length() > 0 && feed->rssurl() != feedurl)
-			continue;
-
-		if (feed->total_item_count() > 0) {
-			if (api) {
+	if (feedurl.empty()) { // Mark all feeds as read
+		if (api) {
+			for (const auto& feed : feedcontainer.feeds) {
```

Comment:
Shouldn't this loop be protected by `feeds_mutex` and `feed->item_mutex`? The old code acquired both locks.

Reply:
I think you're correct.

Reply:
By the way, was introducing those locks done because of some previous bugs with thread syncing?

Reply:
I don't know, but I have it in my TODO to find out because I have second thoughts about separate `feed_lock`s in `controller` and `Feedcontainer`.